pkg/portableresources/backend/controller/createorupdateresource.go

```diff
@@ -119,15 +119,25 @@ func (c *CreateOrUpdateResource[P, T]) Run(ctx context.Context, req *ctrl.Reques
 			return ctrl.Result{}, err
 		}
 	}
-
 	if recipeDataModel.Recipe() != nil {
 		recipeDataModel.Recipe().DeploymentStatus = util.Success
 	}
+
+	datamodel := recipeDataModel.(rpv1.RadiusResourceModel)
```

Comment:
Do we have unit tests for changes made in this file?

Reply:
It's a bit hard to test these with the mocks. That's why probably other fields updated by the code are not validated in the UT as well. This will be covered by the functional test though

Reply:
I think this code should be a part of validator.go where rest of the output processing happens as I mentioned earlier, that will solve the problem of unit testing it as well.

https://github.com/radius-project/radius/pull/6450#discussion_r1353851994